### PR TITLE
Add dto, controller, mapper

### PR DESCRIPTION
**Pull request by KDanil2000**

### CODE REVIEW

File: src/main/java/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import service.BookService;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/api/books")
```
Comment (liliia-ponomarenko) [inline src/main/java/controller/BookController.java]:
```suggestion
@RequestMapping("/books")
```

File: src/main/java/dto/BookDto.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class BookDto {
```
Comment (liliia-ponomarenko) [inline src/main/java/dto/BookDto.java]: Let’s  use record for dtos

File: src/main/java/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -1,27 +1,41 @@
...
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toBook(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
```
Comment (liliia-ponomarenko) [inline src/main/java/service/impl/BookServiceImpl.java]: can be inline variable

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Great! A few minor mistakes  ;)

**Review by Elena-Bruyako — APPROVED** [review]
Good job!

**Pull request merged**